common/lib/dependabot/shared_helpers.rb

```diff
@@ -27,12 +28,16 @@ def self.in_a_temporary_repo_directory(directory = "/",
                                            repo_contents_path = nil,
                                            &block)
       if repo_contents_path
-        path = Pathname.new(File.join(repo_contents_path, directory)).
-               expand_path
+        path = Pathname.new(File.join(repo_contents_path, directory)).expand_path
         reset_git_repo(repo_contents_path)
         # Handle missing directories by creating an empty one and relying on the
         # file fetcher to raise a DependencyFileNotFound error
         FileUtils.mkdir_p(path)
+
+        if (workspace = Dependabot::Workspace.active_workspace)
```

Comment:
```suggestion
        # If a workspace object has been initialised for us, we should prefer to use it so we commit 
        # any changes we make with additional debugging, otherwise we just use the bare commands.
        if (workspace = Dependabot::Workspace.active_workspace)
```

I think this is worth a comment so there's a breadcrumb to look for it being assigned, etc.